scvi/models/vae_fish.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+"""Main module."""
+
+import torch
+import torch.nn.functional as F
+from torch.distributions import Normal, Poisson, kl_divergence as kl
+
+from scvi.metrics.log_likelihood import log_zinb_positive, log_nb_positive
+from scvi.models.modules import Encoder, DecoderSCVI
+from scvi.models.classifier import Classifier
+from scvi.models.utils import one_hot
+from scvi.models import VAE
+
+torch.backends.cudnn.benchmark = True
+
+
+# VAE model
+class VAEF(VAE):
+    r"""Variational auto-encoder model.
+
+    Args:
+        :n_input: Number of input genes for scRNA-seq data.
+        :n_input_fish: Number of input genes for smFISH data
+        :n_batch: Default: ``0``.
+        :n_labels: Default: ``0``.
+        :n_hidden: Number of hidden. Default: ``128``.
+        :n_latent: Default: ``1``.
+        :n_layers: Number of layers. Default: ``1``.
+        :dropout_rate: Default: ``0.1``.
+        :dispersion: Default: ``"gene"``.
+        :log_variational: Default: ``True``.
+        :reconstruction_loss: Default: ``"zinb"``.
+        :reconstruction_loss_fish: Default: ``"poisson"``.
+
+    Examples:
+        >>> gene_dataset_seq = CortexDataset()
+        >>> gene_dataset_fish = SmfishDataset()
+        >>> vae = VAE(gene_dataset_seq.nb_genes, gene_dataset_fish.nb_genes,
+        ... n_labels=gene_dataset.n_labels, use_cuda=True )
+
+    """
+
+    def __init__(self, n_input, indexes_fish_train=None, n_batch=0, n_labels=0, n_hidden=128, n_latent=10,
+                 n_layers=1, n_layers_decoder=1, dropout_rate=0.3,
+                 dispersion="gene", log_variational=True, reconstruction_loss="zinb",
+                 reconstruction_loss_fish="poisson", model_library=False):
+        super(VAEF, self).__init__(n_input, dispersion=dispersion, n_latent=n_latent, log_variational=log_variational,
+                                   reconstruction_loss=reconstruction_loss, n_batch=n_batch, n_labels=n_labels)
+        self.n_input = n_input
+        self.n_input_fish = len(indexes_fish_train)
+        self.indexes_to_keep = indexes_fish_train
+        self.reconstruction_loss_fish = reconstruction_loss_fish
+        self.model_library = model_library
+        # First layer of the encoder isn't shared
+        self.z_encoder = Encoder(n_input, n_hidden, n_hidden=n_hidden, n_layers=1,
```

Comment:
Is there a `z_encoder` already in the parent class? If so, you wouldn't want to instantiate a new one here, I suspect.

Reply:
There is one but it doesn't take the same parameters as input, so if we use the one from the parent class we'd have to specify different parameters in the init (and then we'd lose the interest of inheriting from VAE as we'd have to respecify most features for the initialization)

Reply:
You mean in the call to `super(VAEF, self).__init__(...)` you'd have to specify some different parameters? That's OK --it's an improvement over how it is now. Currently we're getting instantiating `z_encoder` twice, but the parent instance is masked so it's never seen. That's inefficient and confusing for someone new to the code. e.g., if some moves the call to `super(VAEF, self).__init__(...)`  to the end of the `__init__` function, it would totally change the behavior.